ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderStateImpl.java

```diff
@@ -928,6 +928,7 @@ private void yieldLeaderToHigherPriorityPeer() {
                 "is higher than leader's:{} and leader's lastEntry is null",
             this, followerID, currentTerm, followerPriority, leaderPriority);
 
+        server.getLeaderElectionMetrics().onTransferLeader();
```

Comment:
sendStartLeaderElectionToHigherPriorityPeer may return without sending StartLeaderElectionRequest.  Let's move the metrics right before sending the request.
```
+++ b/ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderStateImpl.java
@@ -559,6 +559,7 @@ class LeaderStateImpl implements LeaderState {
     final StartLeaderElectionRequestProto r = ServerProtoUtils.toStartLeaderElectionRequestProto(
         server.getMemberId(), follower, lastEntry);
     CompletableFuture.supplyAsync(() -> {
+      server.getLeaderElectionMetrics().onTransferLeader();
       try {
         StartLeaderElectionReplyProto replyProto = server.getServerRpc().startLeaderElection(r);
         LOG.info("{} received {} reply of StartLeaderElectionRequest from follower:{}",
```